### PR TITLE
Fix transport types

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

**Comment by benma** [comment]
CI fail is a data race likely fixed in https://github.com/digitalbitbox/bitbox-wallet-app/pull/2416/

File: frontends/web/src/utils/transport-android.ts [hunk 1]
```diff
@@ -15,20 +15,20 @@
...
+import { TMsgCallback, TPayload, TQueryPromiseMap } from './transport-common';
 import { runningInAndroid } from './env';
 
 let queryID: number = 0;
 const queryPromises: TQueryPromiseMap = {};
-const currentListeners: Function[] = [];
+const currentListeners: TMsgCallback[] = [];
 
-export function androidCall(query: string) {
+export function androidCall(query: string): Promise<any> {
```
Comment (thisconnect) [inline frontends/web/src/utils/transport-android.ts:25]: Before TS somehow automatically used `Promise<unknown>` which is in theory stricter than `Promise<any>`.

Screen Shot 2023-12-24 at 09 37 00

How about using `Promise<unknown>` here? Although it doesn't matter much I think as we set the correct return type in api/ files. I did a quick test and weblint didn't complain using `Promise<unknown>`.  I slightly prefer unkown here.

File: frontends/web/src/utils/transport-android.ts [hunk 2]
```diff
@@ -15,20 +15,20 @@
...
   return new Promise((resolve, reject) => {
     if (runningInAndroid()) {
       if (typeof window.onAndroidCallResponse === 'undefined') {
         window.onAndroidCallResponse = (
           queryID: number,
-          response: string,
+          response: any,
```
Comment (thisconnect) [inline frontends/web/src/utils/transport-android.ts:31]: same question as before

> Isn't the response always a string?

File: frontends/web/src/utils/transport-qt.ts [hunk 3]
```diff
@@ -51,7 +51,7 @@ async function initTransport() {
   return webChannel;
 }
 
-export function call(query: string) {
+export function call(query: string): Promise<any> {
```
Comment (thisconnect) [inline frontends/web/src/utils/transport-qt.ts:54]: Also here, removing it shows me `Promise<unknown>` which should be stricter.

Screen Shot 2023-12-24 at 10 27 32

File: frontends/web/src/globals.d.ts [hunk 4]
```diff
@@ -14,13 +14,15 @@
...
+
 export declare global {
     interface Window {
         qt?: { webChannelTransport: unknown; };
         android?: {
             call: (queryID: number, query: string) => void;
         }
-        onAndroidCallResponse?: (queryID: number, response: string) => void;
-        onAndroidPushNotification?: (msg: string) => void;
+        onAndroidCallResponse?: (queryID: number, response: any) => void;
```
[the remark this answers is not in the document]
Comment (benma) [reply frontends/web/src/globals.d.ts:25]: No, the string is something is a JSON like `"{ success: true, ... }"` - it is the response of the backend handlers which is always JSON.

File: frontends/web/src/utils/transport-android.ts (hunk 2 above)
Comment (benma) [reply frontends/web/src/utils/transport-android.ts:31]: same answer :P

File: frontends/web/src/utils/transport-android.ts (hunk 1 above)
Comment (benma) [reply frontends/web/src/utils/transport-android.ts:25]: Sure. It also matches the type in `TQueryPromiseMap`

https://github.com/digitalbitbox/bitbox-wallet-app/blob/master/frontends/web/src/utils/transport-common.ts#L53

File: frontends/web/src/utils/transport-qt.ts (hunk 3 above)
Comment (benma) [reply frontends/web/src/utils/transport-qt.ts]: Done, but strange that `apiGet()` and `apiPost()`, which return `call(...)`, have `Promise<any>` and it does not seem to work if changed `Promise<unknown>`.

File: frontends/web/src/globals.d.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (thisconnect) [reply frontends/web/src/globals.d.ts]: you are wright 👍 . I found also where it is converted using JSON.parse:

- https://github.com/digitalbitbox/bitbox-wallet-app/blob/6ee01421939d3fca39d12b73d40b4c7daad392a2/frontends/web/src/utils/transport-websocket.ts#L42
- https://github.com/digitalbitbox/bitbox-wallet-app/blob/6ee01421939d3fca39d12b73d40b4c7daad392a2/frontends/web/src/utils/qwebchannel.js#L79

File: frontends/web/src/utils/transport-qt.ts (hunk 3 above)
Comment (thisconnect) [reply frontends/web/src/utils/transport-qt.ts]: true. I think we could return `Promise<unknown>` for apiGet and apiPost, but then we need to rewrite all api modules and type the return `as Promise<Type>`

```diff
diff --git a/frontends/web/src/api/account.ts b/frontends/web/src/api/account.ts
index 6323214db..1029e95b2 100644
--- a/frontends/web/src/api/account.ts
+++ b/frontends/web/src/api/account.ts
@@ -80,8 +80,8 @@ type TEthAccountCodeAndNameByAddress = SuccessResponse & {
   name: string;
 }
 
-export const getEthAccountCodeAndNameByAddress = (address: string): Promise<TEthAccountCodeAndNameByAddress> => {
-  return apiPost('accounts/eth-account-code', { address });
+export const getEthAccountCodeAndNameByAddress = (address: string) => {
+  return apiPost('accounts/eth-account-code', { address }) as Promise<TEthAccountCodeAndNameByAddress>;
 };
 
 export interface IStatus {
@@ -91,8 +91,8 @@ export interface IStatus {
     offlineError: string | null;
 }
 
-export const getStatus = (code: AccountCode): Promise<IStatus> => {
-  return apiGet(`account/${code}/status`);
+export const getStatus = (code: AccountCode) => {
+  return apiGet(`account/${code}/status`) as Promise<IStatus>;
 };
 ```

But I am not sure if this improves something.

**Review by thisconnect — APPROVED** [review]
LGTM thanks 👍

File: frontends/web/src/utils/transport-qt.ts (hunk 3 above)
Comment (benma) [reply frontends/web/src/utils/transport-qt.ts]: that would look like a step backwards actually

**Pull request merged**